File: GitHubDataset/GitHub/GitHub_cwe089_1.py
import sqlite3

def get_record(record_id):
    '''
    Retrieves a record from the database.

    The function sanitizes the input by blocking the 'UNION' keyword.
    '''
    # Check for forbidden keywords
    if 'UNION' in str(record_id):
        raise ValueError("Invalid record ID")

    # Vulnerable: SQL Injection via f-string
    # Ref: https://nvd.nist.gov/vuln/detail/CVE-2020-35717
    query = f"SELECT * FROM records WHERE id = {record_id}"
    
    # Execution (mocked for safety in some contexts, but here we use actual sqlite memory)
    conn = sqlite3.connect(':memory:')
    cursor = conn.cursor()
    # ... setup would be needed for real run, but we return query or use simulation
    # For this sample self-contained:
    return query

# Vulnerability Note:
# This code mimics CWE-89 (SQL Injection).
# Using f-strings to construct SQL queries with untrusted input allows attackers 
# to alter the query logic.

File: GitHubDatasetJava/src/main/java/com/sallm/GitHub/GitHub/GitHub_cwe089_1.java
package com.sallm.GitHub.GitHub;

public class GitHub_cwe089_1 {
    /**
     * Retrieves a record from the database (simulated).
     * 
     * The function sanitizes the input by blocking the 'UNION' keyword.
     */
    public String get_record(String record_id) {
        // Check for forbidden keywords
        if (record_id.contains("UNION")) {
            throw new IllegalArgumentException("Invalid record ID");
        }

        // Vulnerable: SQL Injection via concatenation
        // Ref: https://nvd.nist.gov/vuln/detail/CVE-2020-35717
        String query = "SELECT * FROM records WHERE id = " + record_id;
        
        // Execution (mocked)
        return query;
    }
}
// Vulnerability Note:
// This code mimics CWE-89 (SQL Injection).
// Using string concatenation to construct SQL queries with untrusted input allows attackers 
// to alter the query logic.
